src/hooks/useExistingRooms.ts

```diff
@@ -0,0 +1,106 @@
+    const invite = useInviter()
+
+    const sessionAccount = wallet?.address
+
+    useEffect(() => {
+        if (!ethereumProvider || !account || !sessionAccount) {
+            return
+        }
+
+        const localRoomIds =
+            ((): string[] | undefined => {
+                try {
+                    return JSON.parse(
+                        localStorage.getItem(StorageKey.RoomIds) || '{}'
+                    )[account]
+                } catch (e) {}
+            })() || []
+
+        dispatch({
+            type: ActionType.SetRoomIds,
+            payload: localRoomIds,
+        })
+
+        // Create a metamask/streamr-client instance to fetch parent identity's streams
+        const providerClient = new StreamrClient({
+            auth: {
+                ethereum: ethereumProvider as any,
+            },
+        })
+
+        async function fn() {
+            const remoteRoomIds: string[] = []
+
+            const streams = await providerClient.listStreams({
+                search: ROOM_PREFIX,
+                operation: 'stream_share' as StreamOperation,
+            })
+
+            await Promise.allSettled(
+                streams.map(async (stream) => {
+                    if (!stream.id.includes(ROOM_PREFIX)) {
+                        return
+                    }
+
+                    try {
+                        const hasSubscribePermission =
+                            await stream.hasUserPermission(
+                                'stream_subscribe' as StreamOperation,
+                                sessionAccount!
+                            )
+
+                        if (!hasSubscribePermission) {
+                            await invite({
+                                invitee: sessionAccount!,
+                                stream,
+                            })
+                        }
+
+                        // Collect up-to-date stream id for clean-up at the end.
+                        remoteRoomIds.push(stream.id)
+
+                        // Append the stream immediately so it shows up ASAP.
+                        dispatch({
+                            type: ActionType.AddRoomIds,
+                            payload: [stream.id],
+                        })
+                    } catch (e) {
+                        // noop
+                    }
+                })
+            )
+
+            // Update the entire list. It's here mostly to eliminate stale rooms (ones that exist
+            // in localStorage but are no longer available online).
+            dispatch({
+                type: ActionType.SetRoomIds,
+                payload: [
+                    // Ids from localStorage stay on top (first loaded, first served). `SetRoomIds`
+                    // action takes care of uniqueness.
+                    ...intersection(localRoomIds, remoteRoomIds),
+                    ...remoteRoomIds,
+                ],
+            })
+        }
+
+        fn()
```

Comment:
Unlike previously, in this version we always go through the list of locally stored ids and see if they're still valid. Someone may have added a room in a different browser, eh?